File: problems/HackerRank/birthday.ts
function birthday(s: number[], d: number, m: number): number {
  let window_size = m;
  let window_start = 0;
  let window_sum = 0;
  let result = 0;

  for (let window_end = 0; window_end < s.length; window_end++) {
    window_sum += s[window_end];

    if (window_end - window_start === window_size - 1) {
      if (window_sum === d) {
        result++;
      }

      // remove element outside window
      window_sum -= s[window_start];

      // move window start forward
      window_start++;
    }
  }

  return result;
}

console.log(birthday([2, 2, 1, 3, 2], 4, 2));

File: problems/HackerRank/week3/birthday.py
def birthday(s, d, m):
    
    window_size = 0
    windown_sum = 0
    window_start = 0
    ans = 0
    
    for i in range(len(s)):
        window_size += 1
        windown_sum += s[i]
        
        if window_size == m:
            if windown_sum == d:
                ans += 1
            
            windown_sum -= s[window_start]
            window_start += 1
            window_size -= 1
        
    return ans